internal/agentcfg/elasticsearch.go

```diff
@@ -240,6 +240,8 @@ func (f *ElasticsearchFetcher) refreshCache(ctx context.Context) (err error) {
 		}
 	}
 
+	f.clearScroll(ctx, scrollID)
```

Comment:
I might be missing some context here but if the single page refresh reguarly errors then we won't be clearing the scroll which might lead to too many scrolls. I think a better way to handle this would be to create a method (something like `doScroll`) with the `for` loop and a defer call to clear scroll.